### PR TITLE
Fix Metrowerks assembler version detection

### PR DESCRIPTION
**Pull request by Nomura-RH**
As per title, this ensures proper version detection for the Metrowerks assembler, as the catch-all regex used by most compilers does not work for it, and I dare not touch that thing.

If I may give my two cents, I think it would be better to add a find_version() method to the base Compiler class, and have each compiler implement it.

### CODE REVIEW

File: mesonbuild/compilers/detect.py [hunk 1]
```diff
@@ -1254,6 +1254,10 @@ def detect_nasm_compiler(env: 'Environment', for_machine: MachineChoice) -> Comp
             env.coredata.add_lang_args(comp_class.language, comp_class, for_machine, env)
             return comp_class([], comp, version, for_machine, info, cc.linker, is_cross=is_cross)
         elif 'Metrowerks' in output or 'Freescale' in output:
+            mwas_ver_match = re.search(r'Version (\d+)\.(\d+)\.?(\d+)? build (\d+)', output)
+            assert mwas_ver_match is not None, 'for mypy'  # because mypy *should* be complaning that this could be None
```
Comment (tristan957) [inline mesonbuild/compilers/detect.py:1258]: Should this be a raised exception instead or do other code paths also assert the regexes work? `complaining` is misspelled

**Comment by tristan957** [comment]
@Nomura-RH did you mean to close this?

**Comment by Nomura-RH** [comment]
> @Nomura-RH did you mean to close this?

Yes, this didn't work as expected, and I also wanted to submit a different PR, so I had to sync my fork to the main repo, which automatically closed the PR.

**Comment by QuLogic** [comment]
> Yes, this didn't work as expected, and I also wanted to submit a different PR, so I had to sync my fork to the main repo, which automatically closed the PR.

Next time, be sure to create a new branch for your PRs; this will reduce such conflicts in the future.

**Pull request closed without merging**